lib/base/utility.hpp

```diff
@@ -157,6 +174,72 @@ class Utility
 	static boost::thread_specific_ptr<unsigned int> m_RandSeed;
 };
 
+#ifndef _WIN32
+template<size_t exceptAmount>
+inline void Utility::CloseAllFDs(const std::array<int, exceptAmount>& except)
+{
+	CloseAllFDs(except, [](int) {});
```

Comment:
Can't this just be a default argument?

Reply:
No, doesn’t compile.

Reply:
It would if `OnClose` wasn't a template parameter but `std::function<void(int)>` was used for the callback parameter type.